orcaexample/kinetics.py

```diff
@@ -0,0 +1,137 @@
+        init_orca_context(memory="12g", cores=8)
+    elif args.cluster_mode.startswith("yarn"):
+        if args.cluster_mode == "yarn-client":
+            init_orca_context(cluster_mode="yarn-client")
+        elif args.cluster_mode == "yarn-cluster":
+            init_orca_context(cluster_mode="yarn-cluster", memory=args.executor_memory, driver_memory=args.driver_memory)
+    elif args.cluster_mode == "spark-submit":
+        init_orca_context(cluster_mode="spark-submit")
+
+# _VALID_TYPES = {tuple, list, str, int, float, bool}
+
+# from fvcore.common.config import CfgNode
+
+# def convert_to_dict(cfg_node, key_list=[]):
+#     """ Convert a config node to dictionary """
+#     if not isinstance(cfg_node, CfgNode):
+#         if type(cfg_node) not in _VALID_TYPES:
+#             print("Key {} with value {} is not a valid type; valid types: {}".format(
+#                 ".".join(key_list), type(cfg_node), _VALID_TYPES), )
+#         return cfg_node
+#     else:
+#         cfg_dict = dict(cfg_node)
+#         for k, v in cfg_dict.items():
+#             cfg_dict[k] = convert_to_dict(v, key_list + [k])
+#         return cfg_dict
+
+class WrappedDataLoader(DataLoader):
+    def __init__(self, loader, func) -> None:
+        data_loader_args = {
+            "dataset": loader.dataset,
+            "batch_size": loader.batch_size,
+            "shuffle": False,
+            "num_workers": loader.num_workers,
+            "collate_fn": func(loader.collate_fn),
+            "pin_memory": loader.pin_memory,
+            "drop_last": loader.drop_last,
+            "timeout": loader.timeout,
+            "worker_init_fn": loader.worker_init_fn
+        }
+        super().__init__(**data_loader_args)
+
+def reduceWrapper(func):
+    def reduceIterElement(batch):
+        batch=func(batch)
+        assert len(batch)==5, "Should yield inputs, labels, index, time, meta in dataloader"
+        return batch[0], batch[1]
+    return reduceIterElement
+
+def train_loader_creator(config, batch_size):
+    train_loader = loader.construct_loader(config, "train")
+    loader.shuffle_dataset(train_loader, 0)
+    return WrappedDataLoader(train_loader, reduceWrapper)
+    # return train_loader
+
+def model_creator(config):
+    return build_model(config)
+
+def optim_creator(model, config):
+    return optim.construct_optimizer(model, config)
+
+def loss_creator(config):
+    return losses.get_loss_func(config.MODEL.LOSS_FUNC)(reduction="mean")
+
+cfg = load_config(args, args.cfg_files)
+
+loss_fun = losses.get_loss_func(cfg.MODEL.LOSS_FUNC)(reduction="mean")
+
+if args.backend == "bigdl":
+    net = model_creator(cfg)
+    optimizer = optim_creator(model=net, config=cfg)
+    orca_estimator = Estimator.from_torch(model=net,
+                                          optimizer=optimizer,
+                                          loss = loss_fun,
+                                          backend=args.backend,
+                                          config=cfg)
```

Comment:
may we extend config type to NameSpace type like CfgNode object here, intead of only dict? @hkvision 
to achieve this we change
`worker_config = self.config.copy()` to 
```
        from copy import copy
        worker_config = copy(self.config)
```